ui/app/routes/vault/cluster/secrets/backend/secret-edit.js

```diff
@@ -98,53 +97,89 @@ export default Route.extend(UnloadModelRoute, {
     if (modelType === 'pki-certificate') {
       secret = secret.replace('cert/', '');
     }
-    return hash({
-      secret: this.store
-        .queryRecord(modelType, { id: secret, backend })
-        .then(secretModel => {
-          if (modelType === 'secret-v2') {
-            let targetVersion = parseInt(params.version || secretModel.currentVersion, 10);
-            let version = secretModel.versions.findBy('version', targetVersion);
-            // 404 if there's no version
-            if (!version) {
-              let error = new DS.AdapterError();
-              set(error, 'httpStatus', 404);
-              throw error;
-            }
-            secretModel.set('engine', backendModel);
-
-            return version.reload().then(() => {
-              secretModel.set('selectedVersion', version);
-              return secretModel;
-            });
-          }
-          return secretModel;
-        })
-        .catch(err => {
-          //don't have access to the metadata, so we'll make
-          //a stub metadata model and try to load the version
-          if (modelType === 'secret-v2' && err.httpStatus === 403) {
-            let secretModel = this.store.createRecord('secret-v2');
-            secretModel.setProperties({
-              engine: backendModel,
-              id: secret,
-              // so we know it's a stub model and won't be saving it
-              // because we don't have access to that endpoint
-              isStub: true,
-            });
-            let targetVersion = params.version ? parseInt(params.version, 10) : null;
-            let versionId = targetVersion ? [backend, secret, targetVersion] : [backend, secret];
-            return this.store
-              .findRecord('secret-v2-version', JSON.stringify(versionId), { reload: true })
-              .then(versionModel => {
-                secretModel.set('selectedVersion', versionModel);
-                return secretModel;
-              });
-          }
+    let secretModel;
+
+    let capabilities = this.capabilities(secret);
+    try {
+      secretModel = await this.store.queryRecord(modelType, { id: secret, backend });
+    } catch (err) {
+      if (err.httpStatus === 403 && (modelType === 'secret-v2' || modelType === 'secret')) {
+        await capabilities;
+        // can't read the path and don't have update permission, so re-throw
+        if (!capabilities.get('canUpdate') && modelType === 'secret') {
```

Comment:
Are you using the `.get('canUpdate')` syntax here to prevent an exception from being thrown? I thought we were able to access properties directly safely now?

Reply:
Nice catch 🦅 👀 ! This is actually here because without it I got a very detailed message that I do indeed need to use `get`- it's one of the edge cases with native getters where ember's get is still needed because the capabilities object is a promise proxy which is a primitive ember provides. https://api.emberjs.com/ember/release/classes/PromiseProxyMixin